src/app/(post-login)/submission/components/ProgramMenu.tsx

```diff
@@ -74,101 +135,51 @@ const parseGQLResp = (data: SideMenuQuery) => {
 	};
 };
 
-const ProgramMenu = ({ searchQuery }: { searchQuery: string }) => {
-	const params = useParams();
+const MenuContent = ({ programName }: { programName: string }) => {
 	const pathname = usePathname();
-	const activeProgramName = typeof params.shortName !== 'string' ? '' : params.shortName;
+	const pathnameLastSegment = pathname.split('/')[-1];
 
-	const pathnameLastSegment = pathname.split('/').findLast(Boolean);
+	const { isDisabled: isSubmissionSystemDisabled } = useSubmissionSystemStatus();
 
-	const {
-		data: gqlData,
-		loading,
-		error,
-	} = useQuery(SIDEMENU_PROGRAMS, {
+	const { data: gqlData } = useQuery(SIDEMENU_PROGRAM_STATUS, {
 		variables: {
-			activeProgramName,
+			activeProgramName: programName,
 			filters: defaultClinicalEntityFilters,
 		},
 	});
 
-	const data = gqlData ? parseGQLResp(gqlData) : null;
-
-	const [activeProgramIndex, setActiveProgramIndex] = useState(-1);
-
-	const programs = gqlData?.programs?.filter(notNull) || [];
-	const filteredPrograms = !searchQuery.length
-		? programs
-		: programs.filter(({ shortName }) => shortName.search(new RegExp(searchQuery, 'i')) > -1);
-
-	const { isDisabled: isSubmissionSystemDisabled } = useSubmissionSystemStatus();
-
-	const setActiveProgram =
-		(index: number): MouseEventHandler =>
-		() =>
-			setActiveProgramIndex(index);
-
-	if (loading) return <Loader />;
-	if (error || data === null) notFound();
-
-	const { clinicalRegistrationHasError, clinicalRegistrationInProgress } = data;
+	const programStatusData = parseProgramStatusGQLResp(gqlData);
 
 	return (
 		<>
-			<Link
-				href="/submission"
-				css={css`
-					text-decoration: none !important;
-				`}
-			>
+			{/** Register Samples */}
+			<Link href={`/submission/program/${programName}/registration`}>
 				<MenuItem
-					level={2}
-					content="All Programs"
-					onClick={setActiveProgram(-1)}
-					selected={pathname === '/submission'}
+					level={3}
+					content={
+						<StatusMenuItem>
+							Register Samples
+							{isSubmissionSystemDisabled ? (
```

Comment:
Just my opinion but I feel like we should move this logic out of render into its own variable given its a nested ternary checking several different conditions

`const StatusIcon = isSubmissionSystemDisabled ? (
		<Icon name="lock" fill="accent3_dark" width="15px" />
	) : programStatusData?.clinicalRegistrationHasError ? (
		<Icon name="exclamation" fill="error" width="15px" />
	) : programStatusData?.clinicalRegistrationInProgress ? (
		<Icon name="ellipses" fill="warning" width="15px" />
	) : null;`
	
	`<StatusMenuItem>
	    Register Samples
	    {StatusIcon}`

Reply:
https://github.com/icgc-argo/rdpc-ui/pull/108/commits/5312b370c5a6658f0abbf8f41ff5a26a95a64338

Can't move it out of `render` because it depends on state but I moved it to it's own variable